File: javascript/268.missing-number.js
/*
 * @lc app=leetcode id=268 lang=javascript
 *
 * [268] Missing Number
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
  const n = nums.length;
  let sum = (n * (n + 1)) >> 1;
  for (const num of nums) {
    sum -= num;
  }
  return sum;
};
// @lc code=end

File: rust/268.missing-number.rs
/*
 * @lc app=leetcode id=268 lang=rust
 *
 * [268] Missing Number
 */

// @lc code=start
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let l = nums.len() as i32;
        let mut sum = l * (l + 1) / 2;
        for n in nums {
            sum -= n;
        }
        return sum;
    }
}
// @lc code=end
